iXBRLViewerInterface.py

```diff
@@ -0,0 +1,64 @@
+# -*- coding: utf-8 -*-
+"""
+:mod:`EdgarRenderer.Cube`
+~~~~~~~~~~~~~~~~~~~
+Edgar(tm) Renderer was created by staff of the U.S. Securities and Exchange Commission.
+Data and content created by government employees within the scope of their employment
+are not subject to domestic copyright protection. 17 U.S.C. 105.
+
+Interface to Arelle.org iXBRLViewerPlugin
+
+Subclasses operation for use as Python Classes (not plugin operation) under EdgarRenderer workflow
+"""
+
+import io, os, sys, zipfile
+from arelle.PluginManager import pluginClassMethods
+from arelle import PluginManager
+
+_iXBRLViewerPlugin = None
+_iXBRLViewer_plugin_info = None
+
+STUB_NAME = "ixbrlviewer.xhtml"
+JS_PATH = "/ixviewer-arelle/ixbrlviewer-1.4.30.js"
```

Comment:
There's a potential versioning issue here. Version 1.4.30 of the ixbrl-viewer javascript is hard coded here, but generation depends on the users installed version of the ixbrl-viewer for the Python code. What happens if those two are not compatible?